node/blockchain/views/signed_change_request.py

```diff
@@ -1,15 +1,53 @@
+from rest_framework import status
+from rest_framework.exceptions import ValidationError as DRFValidationError
 from rest_framework.mixins import CreateModelMixin
+from rest_framework.response import Response
+from rest_framework.settings import api_settings
 from rest_framework.viewsets import GenericViewSet
 
-from node.blockchain.serializers.signed_change_request import SignedChangeRequestSerializer
+from node.blockchain.facade import BlockchainFacade
+from node.blockchain.models import Block
+from node.blockchain.serializers.signed_change_request import (
+    NodeDeclarationSignedChangeRequestSerializer, SignedChangeRequestSerializer
+)
+from node.core.exceptions import ValidationError
+from node.core.utils.cryptography import get_signing_key
+from node.core.utils.types import Type
 
 
 class SignedChangeRequestViewSet(CreateModelMixin, GenericViewSet):
     serializer_class = SignedChangeRequestSerializer
+    serializer_classes = {
+        Type.NODE_DECLARATION: NodeDeclarationSignedChangeRequestSerializer,
+    }
+
+    def get_serializer(self, *args, **kwargs):
+        serializer = super().get_serializer(*args, **kwargs)
+        serializer.is_valid(raise_exception=True)
+
+        signed_change_request_type = Type(serializer.validated_data['message']['type'])
+        serializer_class = self.serializer_classes.get(signed_change_request_type)
+        if not serializer_class:
+            raise ValidationError(f'Invalid signed_change_request_type: {signed_change_request_type}')
+
+        return serializer_class(*args, **kwargs)
 
     def perform_create(self, serializer):
-        instance = serializer.save()
+        signed_change_request = serializer.save()
+        blockchain_facade = BlockchainFacade.get_instance()
+        try:
+            Block.objects.add_block_from_signed_change_request(
```

Comment:
what about CV and regular node roles? The should have a different behavior

Reply:
I added stubs. 
[https://thenewboston.atlassian.net/browse/BC-190](https://thenewboston.atlassian.net/browse/BC-190)